server/src/service/BoardService.js

```diff
@@ -65,4 +66,53 @@ export class BoardService extends BaseService {
         await this.boardRepository.save(createBoard);
         return createBoard.id;
     }
+
+    @Transactional()
+    async getDetailBoard(boardId) {
+        const boardDetail = await this.boardRepository
+            .createQueryBuilder('board')
+            .innerJoin('board.creator', 'creator')
+            .leftJoin('board.invitations', 'invitations')
+            .leftJoin('invitations.user', 'user')
+            .leftJoin('board.lists', 'lists')
+            .leftJoin('lists.cards', 'cards')
+            .leftJoin('cards.comments', 'comments')
+            .select([
+                'board',
+                'creator.id',
+                'creator.name',
+                'invitations',
+                'user.id',
+                'user.name',
+                'user.profileImageUrl',
+                'lists',
+                'cards.id',
+                'cards.title',
+                'cards.position',
+                'cards.dueDate',
+            ])
+            .loadRelationCountAndMap('cards.commentCount', 'cards.comments')
+            .where('board.id = :id', { id: boardId })
+            .getOne();
+
+        if (!boardDetail) {
+            throw new EntityNotFoundError();
+        }
+        if (boardDetail && boardDetail.invitations.length) {
+            delete Object.assign(boardDetail, { invitedUsers: boardDetail.invitations })
+                .invitations;
+            boardDetail.invitedUsers = boardDetail.invitedUsers.map((v) => v.user);
+        }
```

Comment:
```suggestion
        if (Array.isArray(boardDetail?.invitations)) {
            boardDetail.invitedUsers = boardDetail.invitations.map((v) => v.user);
            delete boardDetail.invitations;
        }
```
boardDetail의 유무는 위에서 검사하므로 중복 검사 안하셔도 될 듯합니다!ㅎㅎ
delete와 Object.assign을 활용하는 경우는 처음이네요! 나중에 써봐야겠어요👍
이 경우엔 delete 하는 대상이 너무 모호한 것 같아요. Object.assign이 target이 되는 boardDetail을 반환하긴 하지만, 찾아보지 않으면 모르거든요. 직관적으로 무엇을 할당할지, 삭제할지 코드를 작성하면 좀 더 이해하기 쉬울 것 같다는 느낌이 듭니다

Reply:
리뷰사항 반영해보겠습니당 ㅎㅎ 🙃